presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -226,6 +230,10 @@ protected Node visitShowGrants(ShowGrants showGrants, Void context)
 
                 catalogName = qualifiedTableName.getCatalogName();
 
+                accessControl.checkCanShowTables(
```

Comment:
Add a comment here and below that this check is wrong, and we should add a new method `checkCanShowGrants`